Core/MIPS/IR/IRInterpreter.cpp

```diff
@@ -618,45 +618,45 @@ u32 IRInterpret(MIPSState *mips, const IRInst *inst, int count) {
 		case IROp::Mult:
 		{
 			s64 result = (s64)(s32)mips->r[inst->src1] * (s64)(s32)mips->r[inst->src2];
-			memcpy(&mips->lo, &result, 8);
+			memcpy(&mips->loHi, &result, 8);
 			break;
 		}
 		case IROp::MultU:
 		{
 			u64 result = (u64)mips->r[inst->src1] * (u64)mips->r[inst->src2];
-			memcpy(&mips->lo, &result, 8);
+			memcpy(&mips->loHi, &result, 8);
 			break;
 		}
 		case IROp::Madd:
 		{
 			s64 result;
-			memcpy(&result, &mips->lo, 8);
+			memcpy(&result, &mips->loHi, 8);
 			result += (s64)(s32)mips->r[inst->src1] * (s64)(s32)mips->r[inst->src2];
-			memcpy(&mips->lo, &result, 8);
+			memcpy(&mips->loHi, &result, 8);
 			break;
 		}
 		case IROp::MaddU:
 		{
 			s64 result;
-			memcpy(&result, &mips->lo, 8);
+			memcpy(&result, &mips->loHi, 8);
 			result += (u64)mips->r[inst->src1] * (u64)mips->r[inst->src2];
-			memcpy(&mips->lo, &result, 8);
+			memcpy(&mips->loHi, &result, 8);
 			break;
 		}
 		case IROp::Msub:
 		{
 			s64 result;
-			memcpy(&result, &mips->lo, 8);
+			memcpy(&result, &mips->loHi, 8);
 			result -= (s64)(s32)mips->r[inst->src1] * (s64)(s32)mips->r[inst->src2];
-			memcpy(&mips->lo, &result, 8);
+			memcpy(&mips->loHi, &result, 8);
 			break;
 		}
 		case IROp::MsubU:
 		{
 			s64 result;
-			memcpy(&result, &mips->lo, 8);
+			memcpy(&result, &mips->loHi, 8);
 			result -= (u64)mips->r[inst->src1] * (u64)mips->r[inst->src2];
-			memcpy(&mips->lo, &result, 8);
+			memcpy(&mips->loHi, &result, 8);
```

Comment:
any reason why the original code used memcpy here instead of a cast ?

Reply:
Doesn't look like a good reason, `loHi` via union looks cleaner to me indeed.  I think we can kill the memcpys.

-[Unknown]